lib/active_admin/settings_node.rb

```diff
@@ -1,49 +1,30 @@
 module ActiveAdmin
 
   class SettingsNode
-    def initialize(parent = nil)
-      @parent = parent
-    end
-
-    def settings
-      @settings ||= {}
-    end
-
-    delegate :[]=, to: :settings
-
-    protected
-
-    def parent
-      @parent ||= self.class.new
-    end
-
-    def has_key?(name)
-      settings.has_key?(name.to_sym) || @parent && @parent.has_key?(name.to_sym)
-    end
+    class << self
+      def new(superclass = SettingsNode)
+        Class.new(superclass).tap do |klass|
+          superclass.instance_values.keys.each do |name|
+            klass.accessors(name)
+          end
+        end
+      end
```

Comment:
This is the first time that I see the `.new` method of a class being overriden instead of using `initialize`. And if I read this properly, `SettingsNode.new` is not returning an instance of `SettingsNode` but a new class inheriting from it? I can't say I like it.